internal/juju/models.go

```diff
@@ -50,18 +44,18 @@ func (c *modelsClient) getControllerNameByUUID(uuid string) (*string, error) {
 }
 
 // GetByName retrieves a model by name
-func (c *modelsClient) GetByName(name string) (Model, error) {
+func (c *modelsClient) GetByName(name string) (*params.ModelInfo, error) {
```

Comment:
Do we need to check the import function as that also makes use of this function? I saw the tests passed but maybe this needs double checking.

Reply:
The reason it still works, is because the `*params.ModelInfo` struct also has `Name`, `UUID`, and `Type` which are also the same ... so that Model stuct was somewhat redundant

Reply:
👍  Originally I thought we might need a layer of abstraction for the data-model but it only adds complexity at the moment. If it becomes an issue we can revisit.